eng/code-quality-reports/src/main/resources/spotbugs/spotbugs-exclude.xml

```diff
@@ -1,163 +1,264 @@
+    https://github.com/Azure/azure-sdk-for-java/blob/master/eventhubs/data-plane/azure-eventhubs-eph/src/main/java/com/microsoft/azure/eventprocessorhost/PartitionManager.java#L323 -->
+  <Match>
+    <Class name="com.microsoft.azure.eventprocessorhost.PartitionManager$FinalException"/>
+    <Method name="getInner"/>
+    <Bug pattern="BC_UNCONFIRMED_CAST_OF_RETURN_VALUE"/>
+  </Match>
+
+  <!-- Even thought volatile reference to an array exists in protected API surface area changing it will cause another public API surface to change.
+    https://github.com/Azure/azure-sdk-for-java/blob/master/eventhubs/data-plane/azure-eventhubs-eph/src/main/java/com/microsoft/azure/eventprocessorhost/PartitionManager.java#L29 -->
+  <Match>
+    <Class name="com.microsoft.azure.eventprocessorhost.PartitionManager"/>
+    <Field name="partitionIds"/>
+    <Bug pattern="VO_VOLATILE_REFERENCE_TO_ARRAY"/>
+  </Match>
+
+  <!-- Unchecked cast already exists in the public API surface area, they are logically right, so it can be ignored.
+    At AzureStorageCheckpointLeaseManager.java:[line 180,430,446,542,574]
+      https://github.com/Azure/azure-sdk-for-java/blob/master/eventhubs/data-plane/azure-eventhubs-eph/src/main/java/com/microsoft/azure/eventprocessorhost/AzureStorageCheckpointLeaseManager.java#L180
+    At InMemoryLeaseManager.java:[line 148,155,209,241,268]
+      https://github.com/Azure/azure-sdk-for-java/blob/master/eventhubs/data-plane/azure-eventhubs-eph/src/main/java/com/microsoft/azure/eventprocessorhost/InMemoryLeaseManager.java#L155 -->
+  <Match>
+    <Class name="~com\.microsoft\.azure\.eventprocessorhost\.(AzureStorageCheckpointLeaseManager|InMemoryLeaseManager)"/>
+    <Bug pattern="BC_UNCONFIRMED_CAST"/>
+  </Match>
+
+  <!-- Public inner class already exists in the public API surface area even though it should be static.
+    At EventProcessorOptions.java:[lines 234-237]
+      https://github.com/Azure/azure-sdk-for-java/blob/master/eventhubs/data-plane/azure-eventhubs-eph/src/main/java/com/microsoft/azure/eventprocessorhost/EventProcessorOptions.java#L234 -->
+  <Match>
+    <Class name="~com\.microsoft.azure\.eventprocessorhost\.EventProcessorOptions\$(EndOfStreamInitialPositionProvider|StartOfStreamInitialPositionProvider)"/>
+    <Bug pattern="SIC_INNER_SHOULD_BE_STATIC"/>
+  </Match>
+
+  <!-- Known issue in autorest where it will generate a variable with a value of null then use it in a not null check.
+  https://github.com/Azure/autorest.java/issues/328 -->
+  <Match>
+    <Class name="~com\.microsoft\.azure\.batch\.protocol\.implementation\.[\w]+Impl"/>
+    <Bug pattern="RCN_REDUNDANT_NULLCHECK_OF_NULL_VALUE"/>
+  </Match>
+
+  <Match>
+    <Class name="com.microsoft.azure.batch.CreateTasksErrorException"/>
+    <Or>
+      <Field name="failureTaskList"/>
+      <Field name="pendingTaskList"/>
+    </Or>
+    <Bug pattern="SE_BAD_FIELD"/>
+  </Match>
+
+  <!-- No need to implements cloneable. -->
```

Comment:
These last 4 issues are part of our common library. Since it is not shipped yet, I feel like we should fix these rather than suppress them.
This isn't related to eventhub, I don't think this should be part of the PR.
@sima-zhu Are these ones that you're fixing?

Reply:
I only changed the indentation. No serious changes to the content.

Reply:
Ah, I see. The diffs are off because of the extra lines. 👍 Still surprised we have to suppress these. 🤔

Reply:
I am working on Common package spotbugs. These ones related to batch are from someone else's work. Could you please point out which filter rule in common you are talking about?